.github/scripts/check_content.py

```diff
@@ -0,0 +1,57 @@
+import os.path
+from os import walk
+
+
+def check_content() -> int:
+    """
+    Walks through the filetree rooted at the current working directory.
+    For each file that it finds, it validates the file
+    """
+    error_count = 0
+    DIRECTORIES_TO_EXCLUDE = ['.git', '.github', '.idea', 'venv', 'DO NOT COMMIT']
+    FILES_TO_EXCLUDE = ['.DS_Store', '.gitignore']
+
+    for root, dirs, files in walk('.', topdown=True):
+        dirs[:] = [d for d in dirs if d not in DIRECTORIES_TO_EXCLUDE]
+        files[:] = [f for f in files if f not in FILES_TO_EXCLUDE]
+        for file in files:
+            relative_path = os.path.join(root, file)
+            error_count += check_file(relative_path, file)
+    return error_count
+
+
+def check_file(relative_path: str, file: str) -> int:
+    """
+    Check the given file for any issues.
+
+    Writes error messages to the console, and returns the number of errors found
+
+    :param relative_path: The path of the file, including the directory, relative to the vault's root
+    :param file: The name of the file, without any directory
+    :return: The number of errors found in the given file 
+    """
+    errors = 0
+
+    if '.' not in file:
+        print(f'Error:\n  This file has no extension: consider adding ".md" to its name:\n  {relative_path} ')
+        errors += 1
+
+    # Other checks may be added here in future
+    return errors
+
+
+def get_root_of_vault() -> str:
```

Comment:
Missing documentation.
"""
Helper method that returns the root directory of the vault.
"""

Reply:
Thank you - added - and credited....